docs/proposals/observability.md

```diff
@@ -0,0 +1,315 @@
+    TraceContextExtract TraceContext = "extract"
+
+    // TraceContextInject adds a new context to the request, overwriting existing headers, if any.
+    TraceContextInject TraceContext = "inject"
+
+    // TraceContextPropagate updates the existing context (combines extract and inject).
+    TraceContextPropagate TraceContext = "propagate"
+
+    // TraceContextIgnore skips context headers processing.
+    TraceContextIgnore TraceContext = "ignore"
+)
+```
+
+### YAML
+
+Below is an example YAML version of an `ObservabilityPolicy`:
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha1
+kind: ObservabilityPolicy
+metadata:
+  name: example-observability-policy
+  namespace: default
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: example-route
+    sectionName: example-section
+  tracing:
+    ratio: 10
+    context: inject
+    spanName: example-span
+    spanAttributes:
+      attribute1: value1
+      attribute2: value2
+    enable: "on"
+status:
+  ancestors:
+    ancestorRef:
+      group: gateway.networking.k8s.io
+      kind: Gateway
+      name: example-gateway
+      namespace: default
+    conditions:
+      - type: Accepted
+        status: "True"
+        reason: Accepted
+        message: Policy is accepted
+```
+
+and the HTTPRoute it is attached to:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: example-route
+spec:
+  gatewayClassName: nginx
+  listeners:
+  - name: example-section
+    port: 80
+    protocol: HTTP
+    hostname: "*.example.com"
+status:
+  conditions:
+  ...
+  - type: gateway.nginx.org/ObservabilityPolicyAffected # new condition
+    status: "True"
+    reason: PolicyAffected
+    message: Object affected by an ObservabilityPolicy.
+```
+
+### Status
+
+#### CRD Label
+
+According to the [Policy and Metaresources GEP](https://gateway-api.sigs.k8s.io/geps/gep-713/), the `ObservabilityPolicy` CRD must have the `gateway.networking.k8s.io/policy: direct` label to specify that it is a direct policy.
+This label will help with discoverability and will be used by the planned Gateway API Policy [kubectl plugin](https://gateway-api.sigs.k8s.io/geps/gep-713/#kubectl-plugin-or-command-line-tool).
+
+#### Conditions/Policy Ancestor Status
+
+According to the [Policy and Metaresources GEP](https://gateway-api.sigs.k8s.io/geps/gep-713/), the `ObservabilityPolicy` CRD must include a `status` stanza with a slice of Conditions.
+
+The `Accepted` Condition must be populated on the `ObservabilityPolicy` CRD using the reasons defined in the [PolicyCondition API](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1alpha2/policy_types.go). If these reasons are not sufficient, we can add implementation-specific reasons.
+
+One reason for being `not Accepted` would be the fact that the `GatewaySettings` Policy is not configured, which is a requirement in order for the `ObservabilityPolicy` to work.
```

Comment:
Can we define this reason as part of the design?

Reply:
Naming is fun. Let me know what you think about `GatewaySettingsNotSet`.

Reply:
*NginxProxyConfigNotSet